src/Region/RegionHandler.cc

```diff
@@ -665,15 +648,12 @@ void RegionHandler::RemoveFileRequirementsCache(int file_id) {
         std::unique_lock pv_cube_lock(_pv_cube_mutex);
         _pv_preview_cubes.clear();
     } else {
-        // Iterate through requirements and remove those for given file_id
-        for (auto it = _histogram_req.begin(); it != _histogram_req.end();) {
-            if ((*it).first.file_id == file_id) {
-                it = _histogram_req.erase(it);
-            } else {
-                ++it;
-            }
+        // Remove histogram configurations and cache for given file_id
+        for (auto& histogram : _region_histograms) {
+            histogram.second->ClearFileConfigsCache(file_id);
```

Comment:
I suggest unpacking the key and value when iterating over maps -- even if the key is unused, it makes the loop variable name more idiomatic, since it binds only to the value: `for (auto& [key, histogram] : _region_histograms)...`

We could maybe use `_` for unused variables in situations like this. It's already a convention in some languages, and [in C++26 it will be formalised, with additional special properties](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2169r4.pdf).

Reply:
But it would probably be best to leave this as-is here, and create a separate PR later for adjusting the loop style throughout the code (it's not very consistent).